coding-guidelines/coding-guidelines.md

```diff
@@ -0,0 +1,272 @@
+
+## Comments
+Rules:
+- Comments must be prefixed with a space sign.
+- The `*` sign is a exception to the rule.
+
+Examples:
+```JS
+// Single line Comment
+
+//* A Exception to the space rule
+
+/**
+* Block Comment
+*/
+```
+
+## Strings
+ - Strings are defined with double quotes.
+ - Use template strings if variables are used.
+
+Example:
+```JS
+var string = `Hello ${world}!`;
+```
+
+## Variables and Attribute names
+Variable names always use camelCase names, except Constants which use UPPER_CASE.
+One var, let or const per variable and line.
+
+Examples:
+```JS
+var camelCase = "{value}";
+let CamelCase = "{value}";
+const SOME_CONSTANT = "constant value";
+```
+
+## Object Literals
+Object literals must have no space after the '{' or before the '}' signs.
+
+Valid Example
+```JS
+var objectLiteral = {"key": "value", "key2": "value2"};
+
+var objectLiteral = {
+  "key": "value",
+  "key2": "value2",
+  "key3": "value3",
+  "key4": "value4"
+};
+```
+
+
+## Indentation and Line-Breaks
+Rules:
+- Line breaks are only done if needed.
+- Strings are split at the latest possible position.
+  - At the latest whitespace character.
+  - The second line is indented with 2 spaces.
+  ```JS
+    var longString = "A Very long string which is longer then one line, which is 80 characters " +
+      "in this case. This demonstrates how the second line is indented";
+  ```
+- Object literals are kept on one line if possible.
+  - If the line gets to long put each attribute on a single line.
+  - Multi level object literals must be split in to multiple lines.
+    - Each level in the object must have 2 spaces as indentation.
+
+Examples:
+```JS
+test(
+  "A Very long string which is longer then one line, which is 80 characters " +
+    "in this case. This demonstrates how the second line is indented",
+  "Another string parameter.",
+  {and: "a Literal Object", as: "a parameter", type: "short"},
+  {
+    yet: "another long parameter Object",
+    which: "needs to be split in multiple",
+    lines: 3
+  }
+);
+
+test({
+  a: {
+    b: 10
+  },
+  c: 20,
+  d: 30,
+  e: 40,
+  f: 50,
+  g: 60,
+  h: {
+    i: 70
+  }
+});
+```
+
+## Functions
+Rules:
+  - Function blocks must have a space before the definition.
+  - Function must have a space before the opening parenthesis.
+  - Named functions must not have a space before the opening function brackets.
+  - Functions can be on a single line.
+    - If they are there must be a whitespace inside the curly brackets.
+  - Anonymous functions need to have a space before the opening function bracket.
+  - Arrow functions
+    - can be on a single line.
+    - must omit the parameter brackets if only one parameter is present.
+    - must omit curly brackets if only one statement is given.
+    - must have whitespace around the parameter definition
+
+Examples:
+```JS
+function test() {
+  return;
+}
+
+function test() { var a = 10; return a; }
+
+var test = function () { var b = 10; return b; }
```

Comment:
I think this needs a ```;``` at the end.
We did it like that.

Named function don't need it.

A little bit confusing, it is.